### PR TITLE
added FAQ on limits and conscious departure as optimization

### PR DESCRIPTION
**Pull request by FND**
the phrasing is still a bit awkward, so this is merely intended to start the discussion

would appreciate input from @stilkov, @tillsc, @cdent and whoever else might care

aside: I tried to work what I consider a powerful quote from @cdent into this, but it didn't quite fit in. Any ideas where we might put it:

> ROCA is a distillation of ten years of web development experience.

### CODE REVIEW

File: faq.md [hunk 1]
```diff
@@ -18,6 +18,15 @@ to provide a reference for exactly this purpose.
...
 
+### What are the limits?
+
+As with every architectural choice, there are trade-offs to be made. Sometimes
+strict adherence to ROCA principles is not practical or sensible, so it might be
+judicious to consciously depart from this path for selected parts of an
+otherwise ROCA compliant application. Of course one should be conscious of the
+costs associated with such a decision - however, that's no different from other
+optimizations, e.g. database normalization.
+
```
Comment (cdent) [inline faq.md:29]: An example of a situation where ROCA doesn't quite fit _might_ be a good idea here. It depends on to what extent you want this to be helpful for overcoming that last 10% to get to 100%. If you just want to put the warning out there and leave the rest to the rest of the world, that's fine too.

Comment (FND) [inline faq.md:29]: > An example of a situation where ROCA doesn't quite fit might be a good idea here.

I'd considered that, but it seemed tricky without turning this FAQ into a lengthy article...

Comment (youngbrioche) [inline faq.md:29]: :+1:

Comment (stilkov) [inline faq.md:29]: But the answer doesn’t answer the question. I’m not entirely sure this actually really adds anything other than some common sense statement true for any architecture.

Comment (FND) [inline faq.md:29]: > the answer doesn’t answer the question

True, actually. So ... let's change the question. (e.g. "Is ROCA a fascist regime?")

> I’m not entirely sure this actually really adds anything other than some
> common sense statement true for any architecture.

It doesn't, really - however, many people expect things like ROCA to be comprehensive and get all confused when they hit a wall. This is meant to explicitly concede that ROCA is not a silver bullet, thus defusing misguided discussions.

**Comment by FND** [comment]
I've addressed @stilkov's concern in fd92ed3e3d627270291aeb2941bc26bb4963378c (I think) - so if nobody objects, I'll go ahead and merge this.

**Pull request merged**